### ISSUE TITLE
Issue with BigInt.asUintN

### ISSUE THREAD

**Issue by tonygermano** [mozilla/rhino#1573]
Expected:

```javascript
BigInt.asUintN(64, -1n) === BigInt("0xffffffffffffffff")
```

Actual:
```javascript
BigInt.asUintN(64, -1n) === -1n
```

**Comment by gaurav-chaurasia** [comment]
Hi @tonygermano 

I would like to address this issue. According to the documentation, BigInt.asUintN() is supposed to truncate a BigInt value to the specified number of least significant bits and return that value as an unsigned integer. As mentioned in the documentation: [MDN Web Docs on BigInt.asUintN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN).

Based on this documentation, the expected result for BigInt.asUintN(64, -1n) should be the 64-bit unsigned representation of `-1n`, which in hexadecimal is `0xffffffffffffffff`.

However, the actual result suggests that the function may not be correctly handling negative values. Instead of converting -1n to its unsigned 64-bit representation, the function appears to be returning the original signed value, -1n.

**Comment by p-bakker** [comment]
Sounds like a good first issue then 🙂

I'm sure there are tests in the test262 sure right now marked as failing for this (although I haven't looked)

**Comment by p-bakker** [comment]
@gaurav-chaurasia based on your comment, I concluded that you'd try to create a PR to fix this issue. Just checking in to see how things are going

**Comment by tonygermano** [comment]
I've been kind of living in the spec recently. I haven't created the fix yet, but I do know how to fix this one. Part of the reason I started with #1611 was because so many of the BigInt tests were failing due to the lack of Symbol.toPrimitive being implemented correctly.

**Comment by gaurav-chaurasia** [comment]
> @gaurav-chaurasia based on your comment, I concluded that you'd try to create a PR to fix this issue. Just checking in to see how things are going

Yes yes, I'm relatively new to open source and the Rhino engine, so I'm still getting familiar with the codebase and documentation. I've successfully built Rhino locally, and most tests have passed. However, I am not able to reproduce the issue yet as I'm facing some issues while running the Test262 suite.

@p-bakker / @tonygermano I'd really appreciate any suggestions or guidance you might have. Thank you! :)

**Comment by p-bakker** [comment]
What are the issues you're running into running the Test262 suite?
